cli/src/main.rs

```diff
@@ -1,16 +1,108 @@
+use clap::Parser;
 use core::eval;
+use notify::{
+    Config, PollWatcher, RecommendedWatcher, RecursiveMode, Result, Watcher, WatcherKind,
+};
 use parser::parse;
-use std::{env, fs};
+use std::env;
+use std::{fs, path::Path};
+use std::{fs::File, io::Write, time::Duration};
+use termion::{color, style};
 
-// We would likely want to use "clap" and its derive feature to describe
-// the command-line interface with a declarative struct
+#[derive(Parser, Debug)]
+#[command(author, version, about)]
+struct Args {
+    #[arg(index = 1, help = "Path to input file")]
+    input: String,
 
-fn main() {
-    let path: String = env::args().skip(1).take(1).collect();
-    let source = fs::read_to_string(path).expect("Failed to read file");
+    #[arg(index = 2, help = "Path to output file")]
+    output: String,
 
+    #[arg(
+        short = 'w',
+        long = "watch",
+        help = "Watches file and compiles changes"
+    )]
+    w: bool,
+}
+
+fn compile_file(args: &Args) -> Result<()> {
+    let source = fs::read_to_string(&args.input).expect("[ERROR] Failed to read file...");
     let document = parse(&source);
     let output = eval(&document);
 
-    println!("{output}");
+    let mut output_file = File::create(&args.output)?;
+    output_file.write_all(&output.as_bytes())?;
+    Ok(())
+}
+
+fn watch(args: &Args, target: &String) -> Result<()> {
+    let (tx, rx) = std::sync::mpsc::channel();
+    let mut watcher: Box<dyn Watcher> = if RecommendedWatcher::kind() == WatcherKind::PollWatcher {
+        let config = Config::default().with_poll_interval(Duration::from_secs(10));
+        Box::new(PollWatcher::new(tx, config).unwrap())
+    } else {
+        Box::new(RecommendedWatcher::new(tx, Config::default()).unwrap())
+    };
+
+    watcher.watch(Path::new(&args.input), RecursiveMode::Recursive)?;
+
+    for event in rx {
+        print!("{}{}", termion::clear::All, termion::cursor::Goto(1, 1));
+        println!(
+            "{}Recompiling...{}\n",
+            color::Fg(color::Yellow),
+            style::Reset
+        );
+        match event {
+            Ok(_) => match compile_file(args) {
+                Ok(_) => {}
+                Err(_) => println!("[ERROR] File could not compile..."),
```

Comment:
Might be nice to use `eprintln` instead of `println` to print errors to stderr

Reply:
i opted to just return the errors instead